src/SData.cpp

```diff
@@ -511,6 +510,29 @@ PVR_ERROR SData::GetChannels(ADDON_HANDLE handle, bool radio) {
     return PVR_ERROR_NO_ERROR;
 }
 
+PVR_ERROR SData::GetChannelStreamProperties(const PVR_CHANNEL* channel, PVR_NAMED_VALUE* properties, unsigned int* iPropertiesCount)
+{
+  std::string strUrl = "";
+  std::vector<SC::Channel> channels;
+  channels = m_channelManager->GetChannels();
+  for (std::vector<SC::Channel>::iterator Schannel = channels.begin(); Schannel != channels.end(); ++Schannel)
```

Comment:
You have seen my comments regarding string init and for loop syntax in your pvr.filmon PR, havn't you. ;-)